### DISCUSSION TITLE
prediction graphs of Apidra and Novorapid have been mixed up?

### DISCUSSION THREAD

**Discussion by VincAlastor** [NightscoutFoundation/xDrip#3306] [Bug reports]
Is it possible that the bolus prediction graphs of Apidra and Novorapid have been mixed up? (version 2024.01.16 and earlier)

**Comment by Navid200** [comment]
@TheConen

**Comment by TheConen** [comment]
The current values for Apidra are taken from #1388, the values for Novorapid have not been changed since Novorapid has been added to xDrip+ (see 62157698148a526a92051cf22e55b351c87e4fff). 

Current values for Apidra:
"onset": "10",
"peak": "60-180",
"duration": "300"

Current values for Novorapid:
"onset": "10",
"peak": "75",
"duration": "180"

However, some googling shows that the values for Novorapid might have always been wrong. https://www.medsafe.govt.nz/profs/datasheet/n/NovoRapid.pdf and https://ec.europa.eu/health/documents/community-register/2004/200404077828/anx_7828_en.pdf show the values for Novorapid as:
onset: 10-20
peak: 60-180
duration: 180-300
These values are indeed very similar to the current Apidra values.

On Apidra, there seems to be conflicting information. https://my.clevelandclinic.org/health/drugs/13902-injectable-insulin-medications is the current source for the Apidra values, however, https://www.drugs.com/apidra.html, https://canadianinsulin.com/articles/apidra-insulin-peak-time-onset-duration-action/ and https://pubmed.ncbi.nlm.nih.gov/15934890/ suggest these might be more accurate:
onset: 15
peak: 60
duration: 120-240

I'm happy to adjust the values if a consensus can be found on what to adjust them to. From my personal experience as an Apidra user, I can say that Apidra lasts about 2hrs for me and I find the "new" values to be more realistic than the current ones.

**Reply by VincAlastor** [reply]
thank you very much for all the details. I'm also Apidra (Glulisin) user. And i totally agree with your personal experience. Apidra peaks after 60 min and has an overall duration of 120 min (>6 units). But when i use the xdrip Apidra values for one unit it makes a flat curve for 5h. So i have to use the Novorapid profile to get a 2h curve with a peak. Do i need to reinstall the app?
![Screenshot_20240122-221340_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/24430747/0b7d845f-739c-4ebd-a19e-7861ee685867)
![Screenshot_20240122-221436_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/24430747/2b08479e-7024-4a21-a0de-d95963b0bc2d)

**Comment by TheConen** [comment]
No, reinstalling the app won't change anything. Apidra is currently configured to have a duration of 300min = 5h.

@jamorham I know you are hesitant to change features/values that already have existed for quite some time and where users have become accustomed to the way they behave. However, afaik, since #3157 has been merged, there hasn't been a new stable release, only nightly. So I think it would be fine to change the Apidra values. The Novorapid values might require some more discussion. What do you think?

**Comment by simonpickering** [comment]
Would one option here be to allow users to modify the profiles themselves? E.g. if it were possible to shadow the  /app/src/main/res/raw/insulin_profiles.json file with an identically named file that lives in the user accessible XDrip+ folder.

This ought to be easy to try with a change here: https://github.com/NightscoutFoundation/xDrip/blob/1924301b1fcd6a28cb019c2e9981aad698c794e1/app/src/main/java/com/eveningoutpost/dexdrip/insulin/InsulinManager.java#L135

It would make it possible for power users to fiddle with the values, but difficult to do it by accident.

Along a vaguely similar lines, do the values set in Predictive Simulations > Low level prediction values do anything these days?
